nupic/embodied/multitask/modules/gaussian_mlp_module.py

```diff
@@ -418,194 +293,79 @@ def _get_mean_and_log_std(self, *inputs):
-                 preprocess_module_type=None,
-                 preprocess_output_dim=128,
-                 preprocess_kw_percent_on=0.1,
-                 representation_module_type=None,
-                 representation_module_dims=(128, 128),
-                 learn_std=True,
-                 init_std=1.0,
-                 min_std=1e-6,
-                 max_std=None,
-                 std_parameterization='exp',
-                 normal_distribution_cls=Normal
-                 ):
-        # set all the irelevant parameters to None
-        super().__init__(input_dim=input_dim,
-                         output_dim=output_dim,
-                         hidden_sizes=hidden_sizes,
-                         hidden_nonlinearity=None,
-                         hidden_w_init=None,
-                         hidden_b_init=None,
-                         output_nonlinearity=output_nonlinearity,
-                         output_w_init=None,
-                         output_b_init=None,
-                         learn_std=learn_std,
-                         init_std=init_std,
-                         min_std=min_std,
-                         max_std=max_std,
-                         std_hidden_sizes=None,
-                         std_hidden_nonlinearity=None,
-                         std_hidden_w_init=None,
-                         std_hidden_b_init=None,
-                         std_output_nonlinearity=None,
-                         std_output_w_init=None,
-                         std_parameterization=std_parameterization,
-                         layer_normalization=False,
-                         normal_distribution_cls=normal_distribution_cls)
-        self._n_tasks = num_tasks
         self._shared_mean_log_std_network = CustomDendriticMLP(
             input_dim=input_dim,
+            context_dim=context_dim,
             output_sizes=(output_dim, output_dim),
             hidden_sizes=hidden_sizes,
+            layers_modulated=layers_modulated,
             num_segments=num_segments,
-            dim_context=dim_context,
-            kw=kw,
             kw_percent_on=kw_percent_on,
             context_percent_on=context_percent_on,
             weight_sparsity=weight_sparsity,
             weight_init=weight_init,
+            dendrite_weight_sparsity=dendrite_weight_sparsity,
             dendrite_init=dendrite_init,
             dendritic_layer_class=dendritic_layer_class,
-            output_nonlinearity=output_nonlinearity,
+            output_nonlinearity=output_nonlinearity,       
             preprocess_module_type=preprocess_module_type,
             preprocess_output_dim=preprocess_output_dim,
             preprocess_kw_percent_on=preprocess_kw_percent_on,
-            representation_module_type=representation_module_type,
-            representation_module_dims=representation_module_dims,
         )
 
-    def _get_mean_and_log_std(self, x):
+    def _get_mean_and_log_std(self, *inputs):
```

Comment:
It's the former - this method is calling the forward() function. 

It uses *inputs because I can sometimes take in a context (i.e. if I subclass and use dendrites) or not take in a context (i.e. if I subclass and use a MLP).